src/lib/index.ts

```diff
@@ -23,8 +22,43 @@ class ServerlessIamPerFunctionPlugin {
    * @param options
    */
   constructor(serverless: any) {
-    this.provider = 'aws';
     this.serverless = serverless;
+
+    if (this.serverless.service.provider.name !== 'aws') {
+      throw new this.serverless.classes.Error('ses-template plugin supports only AWS');
```

Comment:
Might want to fix the plugin name here :smile:

Reply:
sorry, my bad. fixed

Reply:
thanks for review. let me know if you see any other issues